tests/unit/test_synapse_workload.py

```diff
@@ -22,6 +22,94 @@
 from .conftest import TEST_SERVER_NAME
 
 
+def test_enable_ip_range_whitelist_success(harness: Harness, monkeypatch: pytest.MonkeyPatch):
+    """
+    arrange: set mock container with file.
+    act: update ip_range_whitelist config and call enable_ip_range_whitelist.
+    assert: new configuration file is pushed and ip_range_whitelist is enabled.
+    """
+    config_content = """
+    listeners:
+        - type: http
+          port: 8080
+          bind_addresses:
+            - "::"
+    """
+    text_io_mock = io.StringIO(config_content)
+    pull_mock = Mock(return_value=text_io_mock)
+    push_mock = MagicMock()
+    container_mock = MagicMock()
+    monkeypatch.setattr(container_mock, "pull", pull_mock)
+    monkeypatch.setattr(container_mock, "push", push_mock)
```

Comment:
What is the reason for needing this? Doesn't harness provide a container with a filesystem now so we don't need to mock anymore?

Reply:
@weiiwang01 wrote that feature 🙂

Reply:
But would this be applicable here since the test is mocking the container to check how the pull/push are being called?

Reply:
I'll close this comment since is outdated so new comments can be made in this new version but feel free to re-open it.